File: leetcode-submissions/0345-reverse-vowels-of-a-string/solution.java
class Solution {
    public String reverseVowels(String s) {
        char[] chars = s.toCharArray();
        int endpointer = chars.length - 1;
        for (int i = 0; i < chars.length; i++) {
            if (isVowel(chars[i])) {
                for (int j = endpointer; j > i; j--) {
                    if (isVowel(chars[j])) {
                        char temp = chars[i];
                        chars[i] = chars[j];
                        chars[j] = temp;
                        endpointer = j - 1;
                        break;
                    }
                }
            }
            if (i >= endpointer) {
                break;
            }
        }
        return String.valueOf(chars);
    }

    public boolean isVowel(char c) {
        return c == 'i' || c == 'u' || c == 'o' ||
        c == 'a' ||c == 'e' || c == 'I' || c == 'U' || c == 'O' ||
        c == 'A' ||c == 'E';
    }
}

File: leetcode-submissions/0345-reverse-vowels-of-a-string/solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        result = []
        for i in s:
            result.append(i)
        l, r = 0, len(s) - 1
        vowels = set()
        vowels.add("a")
        vowels.add("i")
        vowels.add("u")
        vowels.add("o")
        vowels.add("e")

        while l < r:
            if result[l].lower() in vowels and result[r].lower() in vowels:
                result[l], result[r] = result[r], result[l]
                l += 1
                r -= 1

            elif result[l].lower() in vowels:
                r -= 1

            elif result[r].lower() in vowels:
                l += 1

            else:
                l += 1
                r -= 1

        return "".join(result)
